troubadix/runner.py

```diff
@@ -91,6 +93,8 @@ def _check_file(self, file_path: Path) -> FileResults:
         )
 
         for plugin_class in self.plugins.file_plugins:
+            plugin_config = self.plugins_config.get(plugin_class.name, {})
+            context.plugin_config = plugin_config
```

Comment:
The intention was to have only the specific plugin config for the plugin running in the file context, since the same context is used by multiple plugins. And then it could only be added at the point where the plugin is decided, but the context has already been created. When editing the context  __init__ I would then pass the entire config dict and then in the plugin itself it would have to get the specific config from all configs.

Reply:
Plugin classes `__init__` can be changed to take the plugin_config as parameter directly and be then seperate from the file_context.